proto/internal/temporal/server/api/dynamicconfig/v1/message.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package temporal.server.api.dynamicconfig.v1;
+option go_package = "go.temporal.io/server/api/dynamicconfig/v1;dynamicconfig";
+
+import "google/protobuf/struct.proto";
+
+message Constraints {
+	string namespace = 1;
+	string namespace_id = 2;
+	string task_queue_name = 3;
+	int32 task_queue_type = 4;
```

Comment:
We should be able to use 
`TaskQueueType` from `temporal/api/enums/v1/task_queue.proto` and `TaskType` from `temporal/server/api/enums/v1/task.proto`.